src/routes/Profile/Profile.tsx

```diff
@@ -55,30 +50,73 @@ const Profile: React.FunctionComponent = () => {
     }
     setCollectionBadges(result);
     setClaimButtonBulkMode('claim');
-    setClaimButtonModes(
-      SEASON_BADGE_VARIANTS[season.id].reduce(
-        (pV, cI) => ({
-          ...pV,
-          [cI as BadgeVariant]: 'claim',
-        }),
-        {} as Record<BadgeVariant, ClaimButtonProps['mode']>,
+    const claimedTodayVariants = SEASON_BADGE_VARIANTS[season.id].filter((badgeVariant) =>
+      result!.find(
+        ({ claimedAt, variant }) =>
+          variant === badgeVariant &&
+          claimedAt &&
+          DateTime.now().startOf('day').valueOf() <= claimedAt &&
+          claimedAt <= DateTime.now().endOf('day').valueOf(),
+      ),
+    );
+    const claimedInThePastVariants = SEASON_BADGE_VARIANTS[season.id].filter((badgeVariant) =>
+      result!.find(
+        ({ claimedAt, variant }) =>
+          variant === badgeVariant &&
+          claimedAt &&
+          claimedAt < DateTime.now().startOf('day').valueOf(),
       ),
     );
+    const notClaimedVariants = SEASON_BADGE_VARIANTS[season.id].filter((badgeVariant) =>
+      result!.find(({ claimedAt, variant }) => variant === badgeVariant && !claimedAt),
+    );
+    setClaimButtonModes((p) => ({
+      ...p,
+      ...getClaimButtonModesForVariants(notClaimedVariants as BadgeVariant[], 'claim'),
+      ...getClaimButtonModesForVariants(claimedTodayVariants as BadgeVariant[], 'claimed'),
+      ...getClaimButtonModesForVariants(claimedInThePastVariants as BadgeVariant[], 'claimedDate'),
```

Comment:
for those that are claimed in the past we want to show the claimed date for those that are today we don't?

Reply:
exactly! This is the UI logic I talked with Brokoli! It kinda doesnt make sense to show the date of claim if it is today.

Reply:
cool, makes sense